fio_plot/fiolib/jsonimport.py

```diff
@@ -1,45 +1,52 @@
 import os
 import sys
 import json
+import logging
 import pprint
 
+logger = logging.getLogger(__name__)
 
-def filter_json_files(settings, filename):
-    """Filter the json files to only those we need.
-    My ambition is to learn to program and do this right one day.
-    """
-    basename = os.path.basename(filename)
-    splitone = str.split(basename, ".")
-    split = str.split(splitone[0], "-")
-    iodepth = int(split[1])
-    numjobs = int(split[2])
 
-    if iodepth in settings["iodepth"] and numjobs in settings["numjobs"]:
-        return filename
+def filter_json_files(settings, filename):
+    """A bit of a slow process, but guarantees that we get legal
+    json files regardless of their names"""
+    with open(filename, 'r') as candidate_file:
+        try:
+            candidate_json = json.load(candidate_file)
+            if candidate_json["fio version"]:
+                job_options = candidate_json["jobs"][0]["job options"]
+                if job_options["rw"] == settings["rw"]:
+                    iodepth = int(job_options["iodepth"])
+                    numjobs = int(job_options["numjobs"])
+                    if iodepth in settings["iodepth"] and numjobs in settings["numjobs"]:
+                        return filename
+            else:
+                logger.debug(f"{filename} does not appear to be a valid fio json output file, skipping")
+        except Exception as e:
+            logger.warning(e)
 
 
 def list_json_files(settings, fail=True):
     """List all JSON files that maches the command line settings."""
-    json_files = []
+    input_directories = []
```

Comment:
I did some variable renaming here, just made it easier for me to understand the code flow